Tasks/DotNetCoreInstallerV1/versionFetcher.ts

```diff
@@ -0,0 +1,293 @@
+                    if (osSuffix.split("-")[0] != "win" || (osSuffix.split("-")[0] == "win" && downloadPackageInfo.name.endsWith(".zip"))) {
+                        return true;
+                    }
+                }
+
+                return false;
+            });
+
+            if (downloadPackageInfoObject) {
+                return true;
+            }
+
+            return false;
+        });
+
+        if (!!downloadPackageInfoObject && downloadPackageInfoObject.url) {
+            tl.debug("Got download URL for platform with rid: " + downloadPackageInfoObject.rid);
+            return downloadPackageInfoObject.url;
+        }
+
+        return "";
+    }
+
+    private setReleasesIndex(): Promise<void> {
+        return this.httpCallbackClient.get(DotNetCoreReleasesIndexUrl)
+            .then((response: HttpClientResponse) => {
+                return response.readBody();
+            })
+            .then((body: string) => {
+                this.releasesIndex = JSON.parse(body);
+                return;
+            })
+            .catch((ex) => {
+                throw tl.loc("ExceptionWhileDownloadOrReadReleasesIndex", ex);
+            });
+    }
+
+    private getVersionChannel(version: string): any {
+        let versionParts: any = new utils.VersionParts(version);
+
+        let channelVersion = `${versionParts.majorVersion}.${versionParts.minorVersion}`;
+        if (versionParts.minorVersion == "x") {
+            var latestChannelVersion: string = `${versionParts.majorVersion}.0`;
+            this.releasesIndex["releases-index"].forEach(channel => {
+                if (utils.versionCompareFunction(channel["channel-version"], latestChannelVersion) > 0) {
+                    latestChannelVersion = channel["channel-version"];
+                }
+            });
+
+            channelVersion = latestChannelVersion;
+        }
+
+        return this.releasesIndex["releases-index"].find(element => {
+            if (element["channel-version"] == channelVersion) {
+                return true
+            }
+        });
+    }
+
+    private getVersionFromChannel(channelInformation: any, version: string, packageType: string, includePreviewVersions: boolean): Promise<VersionInfo> {
+        let versionParts: utils.VersionParts = new utils.VersionParts(version);
+        var releasesJsonUrl: string = channelInformation["releases.json"];
+        var channelReleases: any = null;
+
+        if (releasesJsonUrl) {
+            return this.httpCallbackClient.get(releasesJsonUrl)
+                .then((response: HttpClientResponse) => {
+                    return response.readBody();
+                })
+                .then((body: string) => {
+                    channelReleases = JSON.parse(body).releases;
+                    if (versionParts.minorVersion == "x" || versionParts.patchVersion == "x") {
+
+                        let latestVersion = "0.0.0";
+                        channelReleases.forEach(release => {
```

Comment:
in case of 'x', latest-runtime/latest-sdk not sufficient?

Reply:
As discussed there are cases where it might not suffice